### PR TITLE
Add Relation!IsTotallyOrdered and auxiliary operators.

### PR DESCRIPTION
**Pull request by lemmy**

### CODE REVIEW

**Comment by lemmy** [comment]
Do we need an InfiniteSets module?!

File: modules/Relation.tla [hunk 1]
```diff
@@ -32,6 +32,11 @@ IsAsymmetric(R, S) == \A x,y \in S : ~(R[x,y] /\ R[y,x])
 (***************************************************************************)
 IsTransitive(R, S) == \A x,y,z \in S : R[x,y] /\ R[y,z] => R[x,z]
 
+(***************************************************************************)
+(* Is the relation R comparable over the set S?                            *)
```
Comment (muenchnerkindl) [inline modules/Relation.tla:36]: The wording of the comment sounds a little strange to me: what does it mean for a relation to be "comparable"? Perhaps "Does R relate all elements of set S" or something along these lines?

File: modules/FiniteSetsExt.tla [hunk 2]
```diff
@@ -134,4 +134,43 @@ Choices(Sets) == LET ChoiceFunction(Ts) == { f \in [Ts -> UNION Ts] :
...
+    /\ IsTransitive(op, S)
+    /\ IsAsymmetric(op, S)
+
+(***************************************************************************)
+(* Given the binary operator Op, is the set S totally ordered?             *)
+(***************************************************************************)
+IsTotallyOrdered(op(_,_), S) ==
+    /\ IsPartiallyOrdered(op, S)
+    /\ IsComparable(op, S)
+
```
Comment (muenchnerkindl) [inline modules/FiniteSetsExt.tla:175]: The definitions all look reasonable to me. But why do they appear in module FiniteSetsExt? They do not intrinsically require the set S to be finite, except of course that TLC wouldn't be able to evaluate them over infinite sets. The same can be said for module Relation, which requires finiteness only due to the way the transitive closure is defined there. A cleaner structure (but requiring more changes) would perhaps be to move the above operators to Relation.tla, move TransitiveClosure and the subsequent operators to a new module (perhaps FiniteGraph.tla), and drop the instance FiniteSets from Relation?

Comment (muenchnerkindl) [reply modules/FiniteSetsExt.tla:175]: Sure, that sounds like a good idea!

Comment (lemmy) [reply modules/FiniteSetsExt.tla]: I ended up adding new operators to Relation.

**Comment by muenchnerkindl** [comment]
LGTM, thanks!

**Pull request merged**